src/extension/src/file_handlers/ExtConfigSettingsHandler.py

```diff
@@ -33,37 +33,70 @@ def __init__(self, logger, json_file_handler, config_folder):
-                                seq_no = cur_seq_no
-                            else:
-                                current_file_m_time = os.path.getmtime(os.path.join(self.config_folder, file))
-                                if current_file_m_time > freshest_time:
-                                    freshest_time = current_file_m_time
-                                    seq_no = cur_seq_no
-                    except ValueError:
-                        continue
-            return seq_no
+            # get seq no from settings file
+            return self.__get_seq_no_from_config_settings()
+
         except Exception as error:
             error_message = "Error occurred while fetching sequence number"
             self.logger.log_error(error_message)
             raise
 
+    def __get_seq_no_from_env_var(self):
+        """ Queries the environment variable to get seq_no and verifies if the corresponding <seq_no>.settings file exists """
+        self.logger.log("Fetching and validating sequence number from the environment variable")
+        seq_no = os.getenv(Constants.SEQ_NO_ENVIRONMENT_VAR)
+
+        if seq_no is None:
+            self.logger.log_warning("Sequence number not found in the environment variable.")
+            return None
+
+        settings_file_path = os.path.join(self.config_folder, str(seq_no) + self.file_ext)
+        if os.path.exists(settings_file_path) and os.path.isfile(settings_file_path):
+            self.logger.log("Valid sequence number found from the environment variable. [Sequence no={0}]".format(str(seq_no)))
+            return seq_no
+
+        self.logger.log_error("Configuration settings file not found, for the current sequence number set in environment variable. [Sequence No={0}]".format(str(seq_no)))
+        return -1
+
+    def __get_seq_no_from_config_settings(self):
+        """ Fetches seq no from the most recent *.settings file in config folder """
+        seq_no = None
+        cur_seq_no = None
+        freshest_time = None
+        for subdir, dirs, files in os.walk(self.config_folder):
+            for file in files:
+                try:
+                    if re.match('^\d+' + self.file_ext + '$', file):
+                        cur_seq_no = int(os.path.basename(file).split('.')[0])
+                        if freshest_time is None:
+                            freshest_time = os.path.getmtime(os.path.join(self.config_folder, file))
+                            seq_no = cur_seq_no
+                        else:
+                            current_file_m_time = os.path.getmtime(os.path.join(self.config_folder, file))
+                            if current_file_m_time > freshest_time:
+                                freshest_time = current_file_m_time
+                                seq_no = cur_seq_no
+                except ValueError:
+                    continue
+        return seq_no
```

Comment:
Please add logging - the goal always is to explain what the function did (and experienced in the environment) looking exclusively at logs.